WSRequestHandler.cpp

```diff
@@ -1093,4 +1095,286 @@ void WSRequestHandler::HandleGetRecordingFolder(WSRequestHandler* req)
 
 	req->SendOKResponse(response);
 	obs_data_release(response);
-}
\ No newline at end of file
+}
+
+void WSRequestHandler::HandleGetTextGDIPlusProperties(WSRequestHandler* req)
+{
+
+	const char* itemName = obs_data_get_string(req->data, "source");
+	if (!itemName)
+	{
+		req->SendErrorResponse("invalid request parameters");
+		return;
+	}
+
+	const char* sceneName = obs_data_get_string(req->data, "scene-name");
+	obs_source_t* scene = Utils::GetSceneFromNameOrCurrent(sceneName);
+	if (scene == NULL) {
+		req->SendErrorResponse("requested scene doesn't exist");
+		return;
+	}
+
+	obs_sceneitem_t* sceneItem = Utils::GetSceneItemFromName(scene, itemName);
+	if (sceneItem != NULL)
+	{
+		obs_source_t* sceneItemSource = obs_sceneitem_get_source(sceneItem);
+		const char* sceneItemSourceId = obs_source_get_id(sceneItemSource);
+
+		if (strcmp(sceneItemSourceId, "text_gdiplus") == 0)
+		{
+			obs_data_t* response = obs_source_get_settings(sceneItemSource);
+			obs_data_set_string(response, "source", itemName);
+			obs_data_set_string(response, "scene-name", sceneName);
+			obs_data_set_bool(response, "render",
+				obs_sceneitem_visible(sceneItem));
+
+			obs_sceneitem_release(sceneItem);
+			req->SendOKResponse(response);
+			obs_data_release(response);
+		}
+		else
+		{
+			req->SendErrorResponse("not text gdi plus source");
+		}
+
+	}
+	else
+	{
+		req->SendErrorResponse("specified scene item doesn't exist");
+	}
+
+	obs_source_release(scene);
+
+}
+
+void WSRequestHandler::HandleSetTextGDIPlusProperties(WSRequestHandler* req)
+{
+	if (!req->hasField("source") )
+	{
+		req->SendErrorResponse("missing request parameters");
+		return;
+	}
+
+	const char* itemName = obs_data_get_string(req->data, "source");
+	if (!itemName)
+	{
+		req->SendErrorResponse("invalid request parameters");
+		return;
+	}
+
+	const char* sceneName = obs_data_get_string(req->data, "scene-name");
+	obs_source_t* scene = Utils::GetSceneFromNameOrCurrent(sceneName);
+	if (scene == NULL) {
+		req->SendErrorResponse("requested scene doesn't exist");
+		return;
+	}
+
+	obs_sceneitem_t* sceneItem = Utils::GetSceneItemFromName(scene, itemName);
+	if (sceneItem != NULL)
+	{
+		obs_source_t* sceneItemSource = obs_sceneitem_get_source(sceneItem);
+		const char* sceneItemSourceId = obs_source_get_id(sceneItemSource);
+
+		if (strcmp(sceneItemSourceId, "text_gdiplus") == 0)
+		{
+
+			obs_data_t* settings = obs_source_get_settings(sceneItemSource);
+
+			if (req->hasField("align"))
+			{
+				obs_data_set_string(settings, "align", 
+					obs_data_get_string(req->data, "align"));
+			}
+
+			if (req->hasField("bk_color"))
+			{
+				obs_data_set_int(settings, "bk_color", 
+					(int)obs_data_get_int(req->data, "bk_color"));
```

Comment:
so looking at this closer obs-txt.cpp does the following:
`#define obs_data_get_uint32 (uint32_t)obs_data_get_int`
I might be over thinking as it really doesn't matter what we do on our end as obs-text will cast them as it needs